src/providers/aws.rs

```diff
@@ -0,0 +1,887 @@
+    where
+        P: ProvideAwsCredentials + Send + Sync + 'static,
+        <P as ProvideAwsCredentials>::Future: Send,
+    {
+        let region = region.parse()?;
+        let ec2 = RegionLauncher::connect(region, provider, log)?
+            .make_security_group()?
+            .make_ssh_key()?;
+
+        Ok(ec2)
+    }
+
+    fn connect<P>(
+        region: rusoto_core::region::Region,
+        provider: P,
+        log: slog::Logger,
+    ) -> Result<Self, Error>
+    where
+        P: ProvideAwsCredentials + Send + Sync + 'static,
+        <P as ProvideAwsCredentials>::Future: Send,
+    {
+        debug!(log, "connecting to ec2");
+        let ec2 = rusoto_ec2::Ec2Client::new_with(HttpClient::new()?, provider, region.clone());
+
+        Ok(Self {
+            region,
+            security_group_id: Default::default(),
+            ssh_key_name: Default::default(),
+            private_key_path: Some(
+                tempfile::NamedTempFile::new()
+                    .context("failed to create temporary file for keypair")?,
+            ),
+            outstanding_spot_request_ids: Default::default(),
+            instances: Default::default(),
+            client: Some(ec2),
+            log: Some(log),
+        })
+    }
+
+    fn make_security_group(mut self) -> Result<Self, Error> {
+        let log = self.log.as_ref().expect("RegionLauncher uninitialized");
+        let ec2 = self.client.as_mut().expect("RegionLauncher unconnected");
+
+        // set up network firewall for machines
+        let group_name = super::rand_name("security");
+        trace!(log, "creating security group"; "name" => &group_name);
+        let mut req = rusoto_ec2::CreateSecurityGroupRequest::default();
+        req.group_name = group_name;
+        req.description = "temporary access group for tsunami VMs".to_string();
+        let res = ec2
+            .create_security_group(req)
+            .sync()
+            .context("failed to create security group for new machines")?;
+        let group_id = res
+            .group_id
+            .expect("aws created security group with no group id");
+        trace!(log, "created security group"; "id" => &group_id);
+
+        let mut req = rusoto_ec2::AuthorizeSecurityGroupIngressRequest::default();
+        req.group_id = Some(group_id.clone());
+
+        // icmp access
+        req.ip_protocol = Some("icmp".to_string());
+        req.from_port = Some(-1);
+        req.to_port = Some(-1);
+        req.cidr_ip = Some("0.0.0.0/0".to_string());
+        trace!(log, "adding icmp access to security group");
+        ec2.authorize_security_group_ingress(req.clone())
+            .sync()
+            .context("failed to fill in security group for new machines")?;
+
+        // cross-VM talk
+        req.ip_protocol = Some("tcp".to_string());
+        req.from_port = Some(0);
+        req.to_port = Some(65535);
+        req.cidr_ip = Some("0.0.0.0/0".to_string());
```

Comment:
I feel like I commented on this previously, but this is a much wider security policy than tsunami [currently has](https://github.com/jonhoo/tsunami/blob/96a2f912f445b57b02a217b4a6998837373989b6/src/lib.rs#L397). I think maybe we want a flag that dictates whether external traffic should be accepted.